File: src/interface/Author/Author.ts
import { observable } from "mobx";

export class Author {
    @observable firstName: string;
    @observable lastName: string;

    constructor(
        
        firstName: string,
        lastName: string,
        public id?: number
    ) {
        this.firstName = firstName;
        this.lastName = lastName;
    }
}

File: backend/Models/Course/Author.cs




using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Identity;

namespace visual_studio_code_courses_authors.Models.Course
{
    // Add profile data for application users by adding properties to the ApplicationUser class
    public class Author
    {
        public int id { get; set; }

        [Required, MaxLength(100)]
        public string firstName { get; set; }

        [Required, MaxLength(100)]
        public string lastName { get; set; }

        [Required, DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime created { get; set; }
    }
}


